### DISCUSSION TITLE
Custom Item Meta

### DISCUSSION THREAD

**Discussion by Debuggingss** [Minestom/Minestom#909] [Help] [answered]
For example I want to track how many kills a sword has by putting "kills: 0" into the meta and updating it on each kill. How can I achieve this?

**Comment by DevBlokJochem** [comment]
You can use tags. Tags are values who are given to entitys like items. So for example, you create an tag called kills. Every time you will make a kill with that weapon, you have to set the tag with 1 more than before. The only thing you have to do when you want to get the amount of kill with that weapon is getting the tag from that item.

**Comment by TheMode** [comment, marked as answer]
You would have to use the [Tag API](https://wiki.minestom.net/feature/tags) and have a tag such as
```java
// Save the tag as constant
Tag<Integer> KILL_TRACKER = Tag.Integer("kill");
// Apply the tag to an existing item
item = item.withTag(KILL_TRACKER, 5);

int kill = item.getTag(KILL_TRACKER);
```

**Reply by Debuggingss** [reply]
Thanks!

**Reply by Debuggingss** [reply]
One more question. Is there a simple way to store a map as a tag? Like I want to track
kills_zombie: 0
kills_pig: 0
except there would be 10 different types.

**Reply by TheMode** [reply]
There is currently no simplified solution for that. You will have to make a `Tag.Structure` with your own custom `TagSerializer` (be sure to only serialize immutable objects, so `Map.copyOf` in your case).

The API is still subject to evolution so feel free to play with it and provide feedback ;)
